r10k.yaml.example

```diff
@@ -1,5 +1,79 @@
-:cachedir: '/var/cache/r10k'
-:sources:
-  :plops:
-    remote: 'git@github.com:my-org/org-shared-modules'
-    basedir: '/etc/puppet/environments'
+---
+
+# The cachedir setting controls where cached content, such as mirrored Git
+# repositories, are stored. This location should be persistent, as environments
+# and modules may rely on these files in order to be updated.
+#
+# See https://github.com/puppetlabs/r10k/blob/master/doc/git/cloning-and-mirroring.mkd
+# for more information on how r10k caches Git repositories.
+#
+# The default value is "~/.r10k"
+cachedir: '/var/cache/r10k'
+
+# The sources setting determines what repositories r10k will use when creating
+# Puppet environments. The structure is a map of source names and a map of the
+# source settings.
+sources:
+  # Each source should have a unique name, and preferrably should only use
+  # alphanumeric characters and underscores.
+  operations:
+
+    # The 'remote' setting determines the location of the VCS repository used
+    # for creating environments.
+    remote: 'git@github.com:my-org/org-operations-modules'
+
+    # The 'basedir' setting specifies where environments will be created for
+    # this source. This setting should match the Puppet 'environmentpath'
+    # setting in puppet.conf.
+    #
+    # See http://docs.puppetlabs.com/references/3.7.latest/configuration.html#environmentpath
+    # for more information about the Puppet 'environmentpath' setting.
+    basedir: '/etc/puppetlabs/puppet/environments'
+
+  # One or more sources can be specified; each source is simple another entry
+  # in the sources map.
+  qa:
+    remote: 'git@github.com:my-org/org-qa-modules'
+
+    # Multiple sources can have the same 'basedir' setting, in case two
+    # different groups want to push different sets of environments to the same
+    # Puppet master. If two different sources have the same basedir care should
+    # be exercised to make sure that two sources don't try create multiple
+    # environments with the same name. See the 'prefix' setting for more
+    # information.
+    basedir: '/etc/puppetlabs/puppet/environments'
+
+    # To avoid collisions between multiple sources being created in the same
+    # directory sources can use the prefix setting. Setting prefix to true
+    # prepends the source name and an underscore to each environment, so for
+    # the qa source the "production" branch will create the "qa_production"
+    # branch.
+    #
+    # See https://github.com/puppetlabs/r10k/blob/master/doc/dynamic-environments/configuration.mkd#prefix
```

Comment:
Are you sure you want to point to master? What if the documentation changes in the FOSS version, but a PE 3.8 user is reading this example file?